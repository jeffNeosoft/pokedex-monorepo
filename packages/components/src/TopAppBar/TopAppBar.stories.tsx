import React from "react";
import TopAppBar from "./TopAppBar";
const props={
    color:"#343a40"
}
export default {
  component: <TopAppBar {...props}/>,
  title: "TopAppBar",
};

export const Default = {
    render :(args)=> <TopAppBar {...args}/>,
    args:props
}
