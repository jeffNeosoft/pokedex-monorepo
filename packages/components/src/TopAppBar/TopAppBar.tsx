import * as React from 'react';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import IconButton from '@mui/material/IconButton';
import CatchingPokemonTwoToneIcon from '@mui/icons-material/CatchingPokemonTwoTone';

export default function TopAppBar({color="#343a40"}) {
  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar position="static" sx={{ backgroundColor:color }}>
        <Toolbar>
          <IconButton
            size="medium"
            edge="start"
            color="inherit"
            aria-label="menu"
            sx={{ mr: 2}}
          >
            <CatchingPokemonTwoToneIcon />
          </IconButton>
          <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
            Pokedex
          </Typography>
        </Toolbar>
      </AppBar>
    </Box>
  );
}
