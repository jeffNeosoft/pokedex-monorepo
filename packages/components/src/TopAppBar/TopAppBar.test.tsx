import React from "react";
import { render, screen } from "@testing-library/react";
import TopAppBar from "./TopAppBar";

describe("Navbar render Test case", () => {
	
	it("renders the app bar with correct props", () => {
		const { container } = render(<TopAppBar/>);
		expect(container).toMatchSnapshot();
	});
});
