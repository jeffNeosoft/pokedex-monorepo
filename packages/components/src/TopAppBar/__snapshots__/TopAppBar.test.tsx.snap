// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar render Test case renders the app bar with correct props 1`] = `
<div>
  <div
    class="MuiBox-root css-i9gxme"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1tanbq8-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-134qg7o-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CatchingPokemonTwoToneIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4c4.08 0 7.45 3.05 7.94 7h-4.06c-.45-1.73-2.02-3-3.88-3s-3.43 1.27-3.87 3H4.06C4.55 7.05 7.92 4 12 4z"
              opacity=".3"
            />
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 2c4.08 0 7.45 3.05 7.94 7h-4.06c-.45-1.73-2.02-3-3.88-3s-3.43 1.27-3.87 3H4.06C4.55 7.05 7.92 4 12 4zm2 8c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm-2 8c-4.08 0-7.45-3.05-7.94-7h4.06c.44 1.73 2.01 3 3.87 3s3.43-1.27 3.87-3h4.06c-.47 3.95-3.84 7-7.92 7z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
        >
          Pokedex
        </div>
      </div>
    </header>
  </div>
</div>
`;
